Upgrade to scapy v2.4.0

googlebleh opened the pull request:
Trying to address issue #117 by upgrading the interface for scapy v2.4.0

googlebleh commented:
I've tested `examples/full_rsa_connection_with_application_data.py`, but I'm sure this has poor coverage. Consider this a WIP.

googlebleh commented:
Current build failure looks like a problem with how `setup.py` patches scapy's `config.py`.
As I work on that, should I make a separate PR or just add to this one?

tintinweb commented:
Hi @googlebleh,

appreciate your help.

ad `setup.py` - looks like the `config.py` changed with the new scapy version and thats why the hacky patch breaks the files syntax (seems like a missing linebrk). we're searching for `load_layers = [...]` and lex-patching `, scapy_ssl_tls` at the end not adding/replacing any linebrks (see [setup.py](https://github.com/tintinweb/scapy-ssl_tls/blob/master/setup.py#L104)). should be easy to fix, however maybe there is a more elegant way to add new layers @gpotter2 ?

gpotter2 commented:
We currently have no other elegant way (apart from using a scapy config file) to auto load non scapy layers :/

gpotter2 commented on scapy_ssl_tls/ssl_tls.py:
```diff
@@ -1328,7 +1340,8 @@ def pre_dissect(self, raw_bytes):
             self.guessed_next_layer = TLSRecord
-        self.fields_desc = [PacketListField("records", None, self.guessed_next_layer)]
+        # what a hack. thanks @p-l- for adding __slots__ :(
+        self.fields_desc[:] = [PacketListField("records", None, self.guessed_next_layer)]
```
See my comment

googlebleh commented:
Thanks for both your comments! I've implemented them and will continue on working to pass the integration tests

PS: I understand all the performance implications of `__slots__`. My comment might have been written in a fit of frustration :). I have removed it in my most recent commit.

wasilukm commented on requirements.txt:
```diff
@@ -1,3 +1,3 @@
 pycryptodomex>=3.4
-scapy>=2.2.0,<2.3.3
+scapy==2.4.0
```
Should be `scapy>=2.4.0`?
